lib/v2/reuters/common.js

```diff
@@ -9,31 +9,54 @@ module.exports = async (ctx) => {
     const topic = ctx.params.topic ?? (category === 'authors' ? 'reuters' : '');
     const limit = ctx.query.limit ? parseInt(ctx.query.limit) : 20;
 
-    const rootUrl = 'https://www.reuters.com';
-    const currentUrl = topic ? `${rootUrl}/${category}/${topic}/` : `${rootUrl}/${category}/`;
-    const response = await got(currentUrl);
-    const $ = cheerio.load(response.data);
-
-    let items = $('.media-story-card__body__3tRWy a.media-story-card__heading__eqhp9, a.svelte-pxbp38, a.svelte-11dknnx, a.svelte-e21rsn')
-        .slice(0, limit)
-        .toArray()
-        .map((item) => {
-            item = $(item);
-            item.find('span.visually-hidden__hidden__2qXMW').remove();
+    const MUST_FETCH_BY_TOPICS = ['authors'];
+    const section_id = `/${category}/${topic ? `${topic}/` : ''}`;
+    const { title, description, rootUrl, response } = await (async () => {
+        if (!MUST_FETCH_BY_TOPICS.includes(category)) {
+            const rootUrl = 'https://www.reuters.com/pf/api/v3/content/fetch/articles-by-section-alias-or-id-v1';
+            const response = await got(
+                `${rootUrl}?query={${encodeURI(
+                    JSON.stringify({
+                        offset: 0,
+                        size: limit,
+                        section_id,
+                        website: 'reuters',
+                    }).slice(1, -1)
```

Comment:
Please use the [searchParams option](https://github.com/sindresorhus/got/tree/v11.8.6#url).

Reply:
My method is exactly what the original request looks like:
```
https://www.reuters.com/pf/api/v3/content/fetch/articles-by-section-alias-or-id-v1?query=%7B%22arc-site%22%3A%22reuters%22%2C%22called_from_a_component%22%3Atrue%2C%22fetch_type%22%3A%22collection%22%2C%22id%22%3A%22%2Fworld%2Fchina%2F%22%2C%22offset%22%3A20%2C%22section_id%22%3A%22%2Fworld%2Fchina%2F%22%2C%22size%22%3A9%2C%22sophi_page%22%3A%22*%22%2C%22sophi_widget%22%3A%22topic%22%2C%22uri%22%3A%22%2Fworld%2Fchina%2F%22%2C%22website%22%3A%22reuters%22%7D&d=155&_website=reuters
```

![image](https://github.com/DIYgod/RSSHub/assets/31068367/da2dc762-db87-4d02-9e40-5f096e6f210a)

Reply:
Oh sorry I see what you mean.